bandwidth-file-spec.txt

```diff
@@ -479,95 +621,260 @@
 
       The number of times that the bandwidth measurements for this relay
       failed because of other reasons.
 
-    "bw_mean=" Int SP
+      This KeyValue was added in version 1.1.0 of this specification.
+
+    "bw_mean=" Int
 
       [Zero or one time.]
 
       The measured bandwidth mean for this relay in bytes per second.
 
-    "bw_median=" Int SP
+      This KeyValue was added in version 1.2.0 of this specification.
+
+    "bw_median=" Int
 
       [Zero or one time.]
 
       The measured bandwidth median for this relay in bytes per second.
 
-    "desc_bw_average=" Int SP
+      This KeyValue was added in version 1.2.0 of this specification.
+
+    "desc_bw_average=" Int
 
       [Zero or one time.]
 
       The descriptor average bandwidth for this relay in bytes per second.
 
-    "desc_bw_obs_last=" Int SP
+      This KeyValue was added in version 1.2.0 of this specification.
+
+    "desc_obs_bw_last=" Int
 
       [Zero or one time.]
 
       The last descriptor observed bandwidth for this relay in bytes per
       second.
 
-    "desc_bw_obs_mean=" Int SP
+      This KeyValue was added in version 1.2.0 of this specification.
+
+    "desc_obs_bw_mean=" Int
 
       [Zero or one time.]
 
       The descriptor observed bandwidth mean for this relay in bytes per
       second.
 
+      This KeyValue was added in version 1.2.0 of this specification.
+
+    "relay_recent_measurements_excluded_error_count=" Int
+
+      [Zero or one time.]
+
+      The number of recent relay measurements which were excluded because
+      the measurement was attempted, but failed. Measurements are recent
+      if they are in the last data_period days (5 by default).
+
+      (See the note in section 1.4, version 1.4.0, about excluded relays.)
+
+      This KeyValue was added in version 1.4.0 of this specification.
+
+    "relay_recent_measurements_excluded_near_count=" Int
+
+      [Zero or one time.]
+
+      When all of a relay's recent successful measurements were performed in
+      a period of time that was too short (by default 1 day), the relay is
+      excluded. This KeyValue contains the number of recent successful
+      measurements for the relay that were ignored for this reason.
+
+      (See the note in section 1.4, version 1.4.0, about excluded relays.)
+
+      This KeyValue was added in version 1.4.0 of this specification.
+
+    "recent_measurements_excluded_old_count=" Int
+
+      [Zero or one time.]
+
+      The number of successful measurements for this relay that are too old
+      (more than 5 days, by default).
+
+      (See the note in section 1.4, version 1.4.0, about excluded relays.)
+
+      This KeyValue was added in version 1.4.0 of this specification.
+
+    "recent_measurements_excluded_few_count=" Int
+
+      [Zero or one time.]
+
+      The number of successful measurements for this relay that were ignored
+      because the relay did not have enough successful measurements (fewer
+      than 2, by default).
```

Comment:
This also exclude the ones excluded by "near" and "old", not sure how to explain it.

Reply:
I will try to explain.

Reply:
Fixed in 3c5db50 using a similar explanation to the header keys.